### PR TITLE
Adds @Container support for InVM Arquillian tests

### PR DESCRIPTION
**Pull request by lordofthejars**

### CODE REVIEW

**Comment by lordofthejars** [comment]
I think that for UberJarContainer we have no way of using @Container annotation

.. I find it weird that GitHub allows me to edit your comments... (@bobmcwhirter)

File: arquillian/api/pom.xml [hunk 1]
```diff
@@ -96,6 +96,18 @@
       <artifactId>junit</artifactId>
       <scope>compile</scope>
     </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-all</artifactId>
```
Comment (bobmcwhirter) [inline arquillian/api/pom.xml:101]: can we dependencyManage that version with a ${version.mockito}?  I think we might use it elsewhere also.

**Comment by bobmcwhirter** [comment]
For uber-jar, if we notice the @Container annotation, could we add some AnnotationBasedContainerFactory boiler-plate class and set a property pointing to the test-class?

The AnnotationBasedContainerFactory could possibly then instantiate the test-class, find the method with the annotation, and delegate to it.

**Comment by bobmcwhirter** [comment]
I did the bits for UberjarSimpleContainer support of the same annotation.

Used in the ArqRequestControllerTest, since I couldn't conceive of a way to unit-test it.  Because I suck.

**Comment by lordofthejars** [comment]
I wanted to do it tomorrow morning but thank you.

**Comment by bobmcwhirter** [comment]
Feel free to review. See above note about how I suck. My bits might could
be improved.

## 

Sent from my iPhone

On May 30, 2016, at 4:39 PM, Alex Soto notifications@github.com wrote:

I wanted to do it tomorrow morning but thank you.
El 30 may. 2016 10:25 p. m., "Bob McWhirter" notifications@github.com
escribió:

> I did the bits for UberjarSimpleContainer support of the same annotation.
> 
> Used in the ArqRequestControllerTest, since I couldn't conceive of a way
> to unit-test it. Because I suck.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <
> https://github.com/wildfly-swarm/wildfly-swarm-core/pull/38#issuecomment-222550445
> ,
> or mute the thread
> <
> https://github.com/notifications/unsubscribe/ABcmYd3JvVYG17aOkIo3PG5QIGdiM0QWks5qG0e4gaJpZM4Ip66Q
> 
> .

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
https://github.com/wildfly-swarm/wildfly-swarm-core/pull/38#issuecomment-222552009,
or mute the thread
https://github.com/notifications/unsubscribe/AAA-T5dWNx_Ppe_QyztaO72dc5bg4YlUks5qG0sUgaJpZM4Ip66Q
.

**Pull request merged**